etcdmain/etcd.go

```diff
@@ -283,15 +283,26 @@ func startEtcdOrProxyV2() {
 	// connections.
 	notifySystemd(lg)
 
-	select {
-	case lerr := <-errc:
-		// fatal out on listener errors
-		if lg != nil {
-			lg.Fatal("listener failed", zap.Error(lerr))
-		} else {
-			plog.Fatal(lerr)
+	var timeoutc <-chan time.Time = nil
+	for {
+		select {
+		case lerr := <-errc:
+			// fatal out on listener errors
+			if lg != nil {
+				lg.Error("listener failed", zap.Error(lerr))
+			} else {
+				plog.Error(lerr)
+			}
+			// It may be the case that several listeners have failed at
+			// once. Exiting as soon as the first one fails would
+			// potentially lose valuable diagnostics, so we wait for a
+			// second and log all available errors on errc before exiting.
+			timeoutc = time.After(1 * time.Second)
```

Comment:
What is the intended behavior?
1. exit the program 1 second after the first error is received? In this case we are collecting errors only within 1 second window.

Or

2. before exiting the program, collect all errors as long as they are not received more than 1 second apart.

Reply:
Ah - (1) was my intention but this probably does (2), so I probably need a `if timeoutc == nil` or similar.

Reply:
Yeah I think the code currently does (2). Checking timeoutc before reassigning sounds good.